presto-docs/src/main/sphinx/language/time.rst

```diff
@@ -0,0 +1,57 @@
+========================
+Legacy And New Timestamp
+========================
+
+New ``TIMESTAMP`` and ``TIME`` semantics align the types with the SQL standard.
+See the following sections for details.
+
+.. note::
+
+   The new ``TIMESTAMP`` semantics is still experimental. It's recommended to keep
+   the legacy ``TIMESTAMP`` semantics enabled. You can experiment with the new semantics
+   by enabling in the global configuration or session level. The legacy semantics
+   may be deprecated in a future release.
+
+Enabling/Disabling
+------------------
+
+The legacy ``TIMESTAMP`` semantics can be set to ``true`` or ``false``
+by setting ``deprecated.legacy-timestamp`` in the ``config.properties.``
+
+Additionally, there is a session level property ``legacy_timestamp``
+that can be set.
+
+TIMESTAMP semantic changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Previously, the ``TIMESTAMP`` type described an instance in time in the Presto session's time zone.
+Now, Presto treats ``TIMESTAMP`` values as a set of the following fields representing wall time:
+
+ * ``YEAR OF ERA``
+ * ``MONTH OF YEAR``
+ * ``DAY OF MONTH``
+ * ``HOUR OF DAY``
+ * ``MINUTE OF HOUR``
+ * ``SECOND OF MINUTE`` - as ``DECIMAL(5, 3)``
+
+For that reason, a ``TIMESTAMP`` value is not linked with the session time zone in any way until a time zone is needed explicitly,
+such as when casting to a ``TIMESTAMP WITH TIME ZONE`` or ``TIME WITH TIME ZONE``.
+In those cases, the time zone offset of the session time zone is applied, as specified in the SQL standard.
+
+TIME semantic changes
+~~~~~~~~~~~~~~~~~~~~~
+
+The ``TIME`` type was changed similarly to the ``TIMESTAMP`` type.
+
+TIME WITH TIME ZONE semantic changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Due to compatibility requirements, having ``TIME WITH TIME ZONE`` completely aligned with the SQL standard was not possible yet.
```

Comment:
Can we explain what is different between the new Presto semantics and what is specified in the SQL standard?

Or is this really trying to say that it's not possible to follow the standard when using political time zones because the standard does not have them?

Reply:
The latter. Let's refactor this paragraph when we remove political zones from TIME.